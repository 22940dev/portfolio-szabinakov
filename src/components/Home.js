import React from 'react'

import htmllogo from '../pictures/HTML-logo.png'
import csslogo from '../pictures/CSS-logo.png'
import reactlogo from '../pictures/React-logo.png'
import nodelogo from '../pictures/Nodejs-logo.png'
import jslogo from '../pictures/JS-logo.png'
import mysqllogo from '../pictures/MySQL-logo.png'

import '../styles/Home.css'

const Home = () => {
    return(
        <>
        <div data-testid='introContainer' className='introContainer'>
            <div className='innerDiv'>
            <div data-testid='helloDiv' className='helloDiv'>
                <h1>Hello,</h1>
            </div>
            <div data-testid='myNameDiv' className='myNameDiv'>
                My name is Szabina!
            </div>
            <div data-testid='engineerContainer' className='engineerContainer'>
                and I am a Software Engineer.
            </div>
            <div data-testid='longAbout' className='longAbout'>
                I have just finished my coding course with Manchester codes, which
                I have started in March 2020.
                Learnt HTML, CSS, Javascript, React, Node.js, Express.js, MySQL.<br/>
                Right now I am looking for my first position as a Software Engineer or any exciting projects,
                where I can deepen my skills and gain some new one.<br/>If you have any suggestion please send me a message.  
            </div>
            <div data-testid='logoContainer' className='logoContainer'>
                <a href='https://html.com/' target='blank'>
                <img className='logoImg' src={htmllogo} alt='HTML-logo'/>
                </a>
                <a href='https://www.w3.org/Style/CSS/Overview.en.html' target='blank'>
                <img className='logoImg' src={csslogo} alt='CSS-logo'/>
                </a>
                <a href='https://www.javascript.com/' target='blank'>
                <img className='logoImg' src={jslogo} alt='Javascript-logo'/>
                </a>
                <a href='https://reactjs.org/' target='blank'>
                <img className='logoImg' src={reactlogo} alt='React-logo'/>
                </a>
                <a href='https://nodejs.org/en/' target='blank'>
                <img className='logoImg' src={nodelogo} alt='Nodejs-logo'/>
                </a>
                <a href='https://www.mysql.com/' target='blank'>
                <img className='logoImg' src={mysqllogo} alt='MySQL-logo'/>
                </a>
            </div>
            </div>
        </div>
        </>

    )
}

export default Home