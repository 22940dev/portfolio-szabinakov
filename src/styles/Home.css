.introContainer {
    display: flex;
    flex-direction: column;
    background-color: lavenderblush;
    border: solid black;
    padding-top: 100px;
}

/* Hello */

h1 {
    margin: 0px;
    margin-bottom: 30px;
}

.helloDiv {
    font-size: 45px;
}


/* My name */

.myName {
    margin: 0px;
    margin-bottom: 40px;
}
.myNameDiv {
    display: flex;
    justify-content: flex-end;
    font-size: 26px;
}


/* Intro */

.aboutContainer {
    font-size: 20px;
    text-align: center;
    padding: 0px 10px 0px 9px;
}

