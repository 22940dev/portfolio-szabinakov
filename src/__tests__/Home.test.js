import React from 'react'
import {render} from '@testing-library/react'
import Home from '../components/Home'

describe('Home', () => {
    it('matches the Snapshot', () => {
        const {asFragment} = render(<Home/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders all elements', () => {
        const {getByTestId} = render(<Home/>)

        expect(getByTestId('introContainer')).toHaveClass('introContainer')
        expect(getByTestId('helloDiv')).toHaveClass('helloDiv')
        expect(getByTestId('myNameDiv')).toHaveClass('myNameDiv')
        expect(getByTestId('engineerContainer')).toHaveClass('engineerContainer')
        expect(getByTestId('longAbout')).toHaveClass('longAbout')
        expect(getByTestId('logoContainer')).toHaveClass('logoContainer')
    })
})
