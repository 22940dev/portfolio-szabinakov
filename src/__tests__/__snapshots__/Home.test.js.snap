// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the Snapshot 1`] = `
<DocumentFragment>
  <div
    class="introContainer"
    data-testid="introContainer"
  >
    <div
      class="innerDiv"
    >
      <div
        class="helloDiv"
        data-testid="helloDiv"
      >
        <h1>
          Hello,
        </h1>
      </div>
      <div
        class="myNameDiv"
        data-testid="myNameDiv"
      >
        My name is Szabina!
      </div>
      <div
        class="engineerContainer"
        data-testid="engineerContainer"
      >
        and I am a Software Engineer.
      </div>
      <div
        class="longAbout"
        data-testid="longAbout"
      >
        I have just finished my coding course with Manchester codes, which I have started in March 2020. Learnt HTML, CSS, Javascript, React, Node.js, Express.js, MySQL.
        <br />
        Right now I am looking for my first position as a Software Engineer or any exciting projects, where I can deepen my skills and gain some new one.
        <br />
        If you have any suggestion please send me a message.
      </div>
      <div
        class="logoContainer"
        data-testid="logoContainer"
      >
        <a
          href="https://html.com/"
          target="blank"
        >
          <img
            alt="HTML-logo"
            class="logoImg"
            src="HTML-logo.png"
          />
        </a>
        <a
          href="https://www.w3.org/Style/CSS/Overview.en.html"
          target="blank"
        >
          <img
            alt="CSS-logo"
            class="logoImg"
            src="CSS-logo.png"
          />
        </a>
        <a
          href="https://www.javascript.com/"
          target="blank"
        >
          <img
            alt="Javascript-logo"
            class="logoImg"
            src="JS-logo.png"
          />
        </a>
        <a
          href="https://reactjs.org/"
          target="blank"
        >
          <img
            alt="React-logo"
            class="logoImg"
            src="React-logo.png"
          />
        </a>
        <a
          href="https://nodejs.org/en/"
          target="blank"
        >
          <img
            alt="Nodejs-logo"
            class="logoImg"
            src="Nodejs-logo.png"
          />
        </a>
        <a
          href="https://www.mysql.com/"
          target="blank"
        >
          <img
            alt="MySQL-logo"
            class="logoImg"
            src="MySQL-logo.png"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
